tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -339,5 +588,571 @@ void LongVector::OpTest::testBaseMethod(
 
   // The TestConfig object handles the logic for extracting the shader output
   // based on the op type.
-  VERIFY_SUCCEEDED(TestConfig.verifyOutput(TestResult));
+  VERIFY_SUCCEEDED(TestConfig->verifyOutput(TestResult));
+}
+
+// Helper to fill the shader buffer based on type. Convenient to be used when
+// copying HLSL*_t types so we can copy the underlying type directly instead of
+// the struct.
+template <typename DataTypeT>
+void LongVector::fillShaderBufferFromLongVectorData(
+    std::vector<BYTE> &ShaderBuffer, std::vector<DataTypeT> &TestData) {
+
+  // Note: DataSize for HLSLHalf_t and HLSLBool_t may be larger than the
+  // underlying type in some cases. Thats fine. Resize just makes sure we have
+  // enough space.
+  const size_t NumElements = TestData.size();
+  const size_t DataSize = sizeof(DataTypeT) * NumElements;
+  ShaderBuffer.resize(DataSize);
+
+  if constexpr (std::is_same_v<DataTypeT, HLSLHalf_t>) {
```

Comment:
another early exit opportunity!

imo any time I see

```
void somefunc(...) {
...
if (case_a) {body_a}
else if (case_b) {body_b}
else if (case_c)  {body_c}
else {body_d}
// end of function with nothing after the else
}
```

its almost always easier to read as 

```
void somefunc(...) {
...
if (case_a) {body_a; return; }
if (case_b) {body_b; return; }
if (case_c)  {body_c; return; }
body_d
}
```